Need some help for generating rrd.

@bkiers 

I was trying to use this project to generate a rrd of json with the commands provided in README.md. 
But I encountered:

```
parsing: JSON.g4 ...
Exception in thread "main" java.lang.ExceptionInInitializerError
	at nl.bigo.rrdantlr4.Main.main(Main.java:46)
Caused by: java.lang.NullPointerException: Cannot invoke "javax.script.ScriptEngine.eval(String)" because "nl.bigo.rrdantlr4.DiagramGenerator.ENGINE" is null
	at nl.bigo.rrdantlr4.DiagramGenerator.<clinit>(DiagramGenerator.java:55)
	... 1 more
```

Any suggestions of how I could get around this would be appreciated. Thank you!

I just followed the instructions in the README.md and it worked fine on my
Mac. Could you post more information about your environment and the command
line you used?

-dan

On Fri, Jun 4, 2021 at 3:15 AM Weixuan Hao ***@***.***> wrote:

> I was trying to use this project to generate a rrd of json with the
> commands provided in README.md.
> But I encountered:
>
> parsing: JSON.g4 ...
> Exception in thread "main" java.lang.ExceptionInInitializerError
> 	at nl.bigo.rrdantlr4.Main.main(Main.java:46)
> Caused by: java.lang.NullPointerException: Cannot invoke "javax.script.ScriptEngine.eval(String)" because "nl.bigo.rrdantlr4.DiagramGenerator.ENGINE" is null
> 	at nl.bigo.rrdantlr4.DiagramGenerator.<clinit>(DiagramGenerator.java:55)
> 	... 1 more
>
> Any suggestions of how I could get around this would be appreciated. Thank
> you!
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bkiers/rrd-antlr4/issues/12>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAKNAFLWFFCAVMMHUNSH57DTRB4RJANCNFSM46CHHEMA>
> .
>

--- a/pom.xml
+++ b/pom.xml
@@ -63,7 +63,7 @@
         <dependency>
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
-            <version>18.0</version>
+            <version>[30.0-jre,)</version>
         </dependency>
 
         <dependency>

--- a/pom.xml
+++ b/pom.xml
@@ -16,8 +16,8 @@
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <args4j.version>2.0.29</args4j.version>
         <itextpdf.version>5.5.6</itextpdf.version>
-        <maven.compiler.source>1.6</maven.compiler.source>
-        <maven.compiler.target>1.6</maven.compiler.target>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
     </properties>
 
     <dependencies>
@@ -77,6 +77,19 @@
             <artifactId>fest-assert</artifactId>
             <version>1.4</version>
         </dependency>
+
+        <dependency>
+            <groupId>org.graalvm.js</groupId>
+            <artifactId>js</artifactId>
+            <version>21.1.0</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.graalvm.js</groupId>
+            <artifactId>js-scriptengine</artifactId>
+            <version>21.1.0</version>
+        </dependency>
+
     </dependencies>
 
     <build>

--- a/src/main/java/nl/bigo/rrdantlr4/DiagramGenerator.java
+++ b/src/main/java/nl/bigo/rrdantlr4/DiagramGenerator.java
@@ -38,7 +38,7 @@ public class DiagramGenerator {
 
     // Java's built-in JS interpreter (Rhino).
     private static final ScriptEngineManager MANAGER = new ScriptEngineManager();
-    private static final ScriptEngine ENGINE = MANAGER.getEngineByName("JavaScript");
+    private static final ScriptEngine ENGINE = MANAGER.getEngineByName("graal.js");
 
     // The library used to convert grammar rules to SVG.
     private static final String RAILROAD_SCRIPT = slurp(DiagramGenerator.class.getResourceAsStream("/railroad-diagram.js"));
@@ -46,7 +46,7 @@ public class DiagramGenerator {
 
     // The templates used to create an HTML page from all grammar rules.
     private static final String HTML_TEMPLATE = slurp(DiagramGenerator.class.getResourceAsStream("/template.html"));
-    private static final String HTML_SIMPE_TEMPLATE = slurp(DiagramGenerator.class.getResourceAsStream("/template.simple.html"));
+    private static final String HTML_SIMPLE_TEMPLATE = slurp(DiagramGenerator.class.getResourceAsStream("/template.simple.html"));
     private static final String CSS_TEMPLATE = slurp(DiagramGenerator.class.getResourceAsStream("/template.css"));
 
     // Initialize the JS engine to load the library used to convert the diagram-DSL
@@ -325,7 +325,7 @@ public class DiagramGenerator {
         }
 
         if(simpleHTML) {
-            final String template = HTML_SIMPE_TEMPLATE
+            final String template = HTML_SIMPLE_TEMPLATE
                 .replace("${rows}", rows);
             return addLinks(antlr4GrammarFileName, template);
         }

